membership-attribute-service/app/services/ContributionsStoreDatabaseService.scala

```diff
@@ -0,0 +1,66 @@
+package services
+
+import anorm._
+import com.typesafe.scalalogging.StrictLogging
+import models.ContributionData
+import play.api.db.Database
+import services.OneOffContributionDatabaseService.DatabaseGetResult
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import scalaz.\/
+
+
+trait OneOffContributionDatabaseService {
+  def getAllContributions(identityId: String): DatabaseGetResult[List[ContributionData]]
+
+  def getLatestContribution(identityId: String): DatabaseGetResult[Option[ContributionData]]
+}
+
+object OneOffContributionDatabaseService {
+  type DatabaseGetResult[R] = Future[\/[String, R]]
+}
+
+class PostgresDatabaseService private (database: Database)(implicit ec: ExecutionContext)
+  extends OneOffContributionDatabaseService with StrictLogging {
+
+  private def executeQuery[R](statement: SimpleSql[Row], parser: ResultSetParser[R]): DatabaseGetResult[R] =
+    Future(database.withConnection { implicit conn =>
+      statement.as(parser)
+    })
+      .map(\/.right)
+      .recover { case NonFatal(err) =>
+        \/.left(s"Error querying contributions store. Error: $err")
+      }
+
+  override def getAllContributions(identityId: String): DatabaseGetResult[List[ContributionData]] = {
+    val statement = SQL"""
```

Comment:
Should Identity ID be forced into the format "integer string" to protect against SQL injection? Or written another way: Do we need a type: NumericString, or maybe just use Integer, if that's what's the identity_id type is in the PostGres table?

Reply:
`identity_id` has type `text` - https://github.com/guardian/contributions-platform/blob/master/contributions-store/sql/create-table.sql#L17
Though it looks like our identity ids are always numeric.
I can add some (scala) code to verify that the string contains a numeric value - is that what you mean?

Reply:
Ok, at this point in the code it must be an authenticated user id so I think it's pretty safe as it is